server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -268,6 +288,51 @@ class InteractiveSession(
     }
   }
 
+  private def datanucleusJars(livyConf: LivyConf): ArrayBuffer[String] = {
+    if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
+      // datanucleus jars has already been in classpath in integration test
+      ArrayBuffer.empty
+    } else {
+      val jars = ArrayBuffer.empty[String]
+      val sparkHome = livyConf.sparkHome().get
+      val libdir =
+        if (new File(sparkHome, "RELEASE").isFile) {
+          new File(sparkHome, "lib")
+        } else {
+          new File(sparkHome, "lib_managed/jars")
+        }
+
+      if (libdir.isDirectory()) {
+        jars ++= libdir.listFiles().filter(_.getName.startsWith("datanucleus-"))
+          .map(_.getAbsolutePath)
+      }
+      require(jars.nonEmpty, "datanucleus jars can not be found")
```

Comment:
HiveContext will fail to start when it is using metastore server without datanucleus jars. I suspect this might be spark issue, because from the logs it seems spark will still initialize embedded derby even when I configure to use metastore server. I will look at it later. 
```
16/06/11 08:38:40 INFO MetaStoreDirectSql: Using direct SQL, underlying DB is DERBY
```